### PR TITLE
Adding iisnode-require reference

### PR DESCRIPTION
**Pull request by DomLennonZA**
Allows NodeBB to run out-of-the-box in IISNode and Azure.

### CODE REVIEW

**Comment by julianlam** [comment]
Wow, this is a module built specifically for NodeBB... do you have a repo for it?

**Comment by pitaj** [comment]
Appears to be https://github.com/tjanczuk/iisnode

**Comment by julianlam** [comment]
Yeah, that's the original lib, but does the code here have changes that make it NodeBB specific? (If so, I'm hesitant to merge as then we'd take on the burden of maintaining it).

If it is just a carbon copy of the iisnode repo, then it should refer back to it in the module's `package.json`...

**Comment by DomLennonZA** [comment]
This isn't the entire IISNode plugin. The IISNode plugin referenced here is a plugin for IIS that allows you to run NodeJS in IIS. The issue is that when using this, if you try and use _require.main.require_, so requiring from the main module, it fails. 

The reason for this is that IISNode runs its applications through a module called _interceptor.js_. This module checks if we are running in IISNode first, so anything not running in IISNode will remain the same. If it is in IISNode, it wraps the login for _require_ in a method that works in the environment. 

I haven't created a GitHub repo yet, but I'll make on when I get home.

tl;dr: This isn't IISNode, its a module that lets NodeBB run on IISNode out of the box with no changes required to the code base

**Comment by nodebb-misty** [comment]
Issue closed due to inactivity.

----

*This is an automated message. If you feel this action was in error, please comment on this issue so it can be looked at again*
